cmd/server/main.go

```diff
@@ -19,30 +19,33 @@ import (
 	"os"
 	"os/signal"
 	"syscall"
+	"time"
 
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	flag "github.com/spf13/pflag"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 	"sigs.k8s.io/aws-encryption-provider/pkg/cloud"
 	"sigs.k8s.io/aws-encryption-provider/pkg/connection"
+	"sigs.k8s.io/aws-encryption-provider/pkg/healthz"
 	"sigs.k8s.io/aws-encryption-provider/pkg/logging"
 	"sigs.k8s.io/aws-encryption-provider/pkg/plugin"
 	"sigs.k8s.io/aws-encryption-provider/pkg/server"
 )
 
 func main() {
 	var (
-		healthzPath   = flag.String("healthz-path", "/healthz", "healthcheck path")
-		healthzPort   = flag.String("health-port", ":8080", "healthcheck port")
-		addr          = flag.String("listen", "/var/run/kmsplugin/socket.sock", "GRPC listen address")
-		key           = flag.String("key", "", "AWS KMS Key")
-		region        = flag.String("region", "us-east-1", "AWS Region")
-		kmsEndpoint   = flag.String("kms-endpoint", "", "use this KMS endpoint instead of the one generated by AWS sdk")
-		qpsLimit      = flag.Int("qps-limit", 0, "number of requests per second to allow for KMS API calls (0 to not rate limit)")
-		burstLimit    = flag.Int("burst-limit", 0, "number of tokens that can be consumed in a single call")
-		encryptionCtx = flag.StringToString("encryption-context", nil, "AWS KMS Encryption Context (e.g. 'a=b,c=d')")
-		debug         = flag.Bool("debug", false, "Print debug level logs")
+		healthzPath       = flag.String("healthz-path", "/healthz", "healthcheck path")
+		healthzPort       = flag.String("health-port", ":8080", "healthcheck port")
+		healthzPullPeriod = flag.Duration("healthz-pull-period", 30*time.Second, `period between check kms API health checks`)
```

Comment:
Can we start out with a hard-coded 30s and add a flag later? There are already 9 other flags for this relatively simple component